app/models/service.rb

```diff
@@ -518,6 +520,10 @@ def oidc_configuration
 
   private
 
+  def achieve_as_deleted
+    ::DeletedObject.create!(object: self, owner: account)
```

Comment:
what if `create!` would fail? it would rollback the destroy?

Reply:
it is `after_commit :achieve_as_deleted, on: :destroy`, so if the destroy fails, or even if it is done but it is inside a transaction and rollbacks, then it does not reach here 😄

Reply:
ah you mean this `create!` ... then nothing, it wouldn't rollback, i think 🤷‍♀

Reply:
it shouldn't rollback because it is not inside the same transaction but after_commit

Reply:
ok, you're right.. `after_commit` is being called after the transaction.